src/TransitionGroupContext.js

```diff
@@ -0,0 +1,47 @@
+import * as PropTypes from 'prop-types'
+import React from 'react'
+
+const TransitionGroupContext = React.createContext && React.createContext(null)
+
+export const transitionGroupContextPropType = PropTypes.shape({
+  isMounting: PropTypes.bool.isRequired
+})
+
+const Consumer = (TransitionGroupContext && TransitionGroupContext.Consumer) || class Consumer extends React.Component {
+  static contextTypes = {
+    transitionGroup: PropTypes.object,
+  }
+  render() {
+    return this.props.children(this.context.transitionGroup)
+  }
+}
+
+const Provider = (TransitionGroupContext && TransitionGroupContext.Provider) || class Provider extends React.Component {
+  static childContextTypes = {
+    transitionGroup: transitionGroupContextPropType,
+  }
+  static propTypes = {
+    value: transitionGroupContextPropType
+  }
+  getChildContext() {
+    return { transitionGroup: this.props.value }
+  }
+  render() { return this.props.children }
+}
+
+export function withTransitionGroup(ComponentToWrap) {
```

Comment:
since we are gonna use just the React api's, lets remove this, and go for the new contextType api in Transition maybe? Alternatively keep the HoC but lets use forwardRef as well

Reply:
Actually for a HoC lets use `react-context-toolbox`'s mapContextToProps helper which takes care of this

Reply:
OK, done. See the comment on the PR about some issues with the "react-context-toolbox" module.

Reply:
@jquense Do you still want to go with react-context-toolbox package? contextType looks the best way.  Why HOC?